src/systems/systemDialog.component.tsx

```diff
@@ -27,185 +27,159 @@ import {
   usePostSystem,
 } from '../api/systems';
 
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Controller, useForm } from 'react-hook-form';
+import { RequestType, SystemsSchema } from '../form.schemas';
 import handleIMS_APIError from '../handleIMS_APIError';
-import { trimStringValues } from '../utils';
-
-export type SystemDialogType = 'add' | 'edit' | 'save as';
-
-const getEmptySystem = (): SystemPost => {
-  return {
-    // Here using null for optional values only, so that types for isUpdated parameters
-    // can match
-    name: '',
-    description: null,
-    location: null,
-    owner: null,
-    importance: SystemImportanceType.MEDIUM,
-  } as SystemPost;
-};
 
 export interface SystemDialogProps {
   open: boolean;
   onClose: () => void;
-  type: SystemDialogType;
+  requestType: RequestType;
+  saveAs?: boolean;
```

Comment:
What was the reason for this again? Is there a case we would have a saveAs post? The dialog type itself is save as, edit or add dialogue, not a post or patch dialogue.

Reply:
I changed it because it reads better in the systems component as well it interactive with the zod schemeas better in the systems dialog component. I'm planning on changing all the form dialog to this style